rust-utils/src/launcher_lair_client.rs

```diff
@@ -0,0 +1,331 @@
+#![deny(clippy::all)]
+
+use std::ops::Deref;
+
+use ed25519_dalek::{Keypair, PublicKey, SecretKey};
+use hc_seed_bundle::{LockedSeedCipher, UnlockedSeedBundle};
+use holo_hash::{AgentPubKey, AgentPubKeyB64};
+use holochain_zome_types::prelude::{Signature, ZomeCallUnsigned};
+use lair_keystore_api::{
+    dependencies::{sodoken::BufRead, url::Url},
+    ipc_keystore::ipc_keystore_connect,
+    lair_store::LairEntryInfo,
+    LairClient,
+};
+use pinentry::PassphraseInput;
+use secrecy::{ExposeSecret, SecretString};
+
+use napi::Result;
+
+use crate::types::*;
+
+struct LauncherLairClient {
+    lair_client: LairClient,
+}
+
+impl LauncherLairClient {
+    /// Connect to lair keystore
+    pub async fn new(connection_url: String, passphrase: String) -> Self {
+        let connection_url_parsed = Url::parse(connection_url.deref()).unwrap();
+        let passphrase_bufread: BufRead = passphrase.as_bytes().into();
+
+        let lair_client = ipc_keystore_connect(connection_url_parsed, passphrase_bufread)
+            .await
+            .unwrap();
+
+        Self { lair_client }
+    }
+
+    /// Sign a zome call
+    pub async fn sign_zome_call(
+        &self,
+        zome_call_unsigned_js: ZomeCallUnsignedNapi,
+    ) -> Result<ZomeCallNapi> {
+        let zome_call_unsigned: ZomeCallUnsigned = zome_call_unsigned_js.clone().into();
+        let pub_key = zome_call_unsigned.provenance.clone();
+        let mut pub_key_2 = [0; 32];
+        pub_key_2.copy_from_slice(pub_key.get_raw_32());
+
+        let data_to_sign = zome_call_unsigned.data_to_sign().unwrap();
+
+        let sig = self
+            .lair_client
+            .sign_by_pub_key(pub_key_2.into(), None, data_to_sign)
+            .await
+            .unwrap();
+
+        let signature = Signature(*sig.0);
+
+        let signed_zome_call = ZomeCallNapi {
+            cell_id: zome_call_unsigned_js.cell_id,
+            zome_name: zome_call_unsigned.zome_name.to_string(),
+            fn_name: zome_call_unsigned.fn_name.0,
+            payload: zome_call_unsigned_js.payload,
+            cap_secret: zome_call_unsigned_js.cap_secret,
+            provenance: zome_call_unsigned_js.provenance,
+            nonce: zome_call_unsigned_js.nonce,
+            expires_at: zome_call_unsigned_js.expires_at,
+            signature: signature.0.to_vec(),
+        };
+
+        Ok(signed_zome_call)
+    }
+
+    /// Reads a json file containing the device bundle and the device derivation path,
+    /// then derives the device seed and from it a sub seed at index 1 whose base64 encoded
+    /// public part should equal the initial_host_pub_key field.
+    pub async fn derive_and_import_seed_from_json_file(
```

Comment:
This is the relevant method. The zome_call_signing stuff is only part of this PR because the file got renamed.